src/base/DeltaResolver.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.24;
+
+import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
+import {ImmutableState} from "./ImmutableState.sol";
+
+/// @notice Abstract contract used to sync, send, and settle funds to the pool manager
+/// @dev Note that sync() is called before any erc-20 transfer in `settle`.
+abstract contract DeltaResolver is ImmutableState {
```

Comment:
hm the inheriting structure in the base PositionManager contract now inherits two contracts(SafeCallback and this one) that both also inherit ImmutableState? Any way we can avoid that?

Reply:
maybe we need to expose a virtual function poolManager() that is overridden in the main base contract ?